dev/src/sdlaudiosfxr.cpp

```diff
@@ -417,18 +417,85 @@ void SDLSoundClose() {
     SDL_QuitSubSystem(SDL_INIT_AUDIO);
 }
 
-bool SDLPlaySound(string_view filename, bool sfxr, int vol) {
+int SDLPlaySound(string_view filename, bool sfxr, float vol, int loops) {
     #ifdef __EMSCRIPTEN__
     // Distorted in firefox and no audio at all in chrome, disable for now.
-    return false;
+        return 0;
     #endif
 
     if (!SDLSoundInit())
-        return false;
+        return 0;
 
     auto snd = LoadSound(filename, sfxr);
+    int ch = -1;
     if (snd) {
-        Mix_Volume(Mix_PlayChannel(-1, snd->chunk.get(), 0), vol);
+        ch = Mix_PlayChannel(-1, snd->chunk.get(), loops);
+        if (ch >= 0)
+        {
+            Mix_Volume(ch, (int)(MIX_MAX_VOLUME*vol));
+        }
     }
-    return !!snd;
+    return ++ch; // we return channel numbers 1..8 rather than 0..7
+}
+
+bool SDLHaltSound(int ch) {
+    #ifdef __EMSCRIPTEN__
+    // Distorted in firefox and no audio at all in chrome, disable for now.
+        return false;
+    #endif
+
+    if (!SDLSoundInit())
+        return false;
+
+    ch--; // SDL_Mixer enmerates channels from 0, Lobster from 1
+    Mix_Resume(ch); // fix SDL bug not clearing the paused flag on halt by resuming (even if not paused) right before halt
+    Mix_HaltChannel(ch);
+    return true;
 }
+
+bool SDLPauseSound(int ch) {
+    #ifdef __EMSCRIPTEN__
```

Comment:
lets put this `#ifdef` inside `SDLSoundInit` so its not repeated in every function.

Reply:
Ok